src/ol/interaction/Modify.js

```diff
@@ -376,14 +408,27 @@ class Modify extends PointerInteraction {
   }
 
   /**
-   * @param {import("../MapBrowserEvent.js").default} evt Map browser event
+   * @param {import("../MapBrowserEvent.js").default} evt Map browser event.
+   * @param {Array<Array<SegmentData>>} segments The segments subject to modification.
    * @private
    */
-  willModifyFeatures_(evt) {
-    if (!this.modified_) {
-      this.modified_ = true;
+  willModifyFeatures_(evt, segments) {
+    if (!this.featuresBeingModified_) {
+      this.featuresBeingModified_ = new Collection();
+      const features = this.featuresBeingModified_.getArray();
+      for (let i = 0, ii = segments.length; i < ii; ++i) {
+        const feature = segments[i][0].feature;
+        if (features.indexOf(feature) === -1) {
+          this.featuresBeingModified_.push(feature);
+        }
+      }
+
       this.dispatchEvent(
-        new ModifyEvent(ModifyEventType.MODIFYSTART, this.features_, evt)
+        new ModifyEvent(
+          ModifyEventType.MODIFYSTART,
+          new Collection(features),
```

Comment:
This should be `this.featuresBeingModified_` instead of a new Collection.

Reply:
Thanks, fixed.